Sources/Build/BuildOperation.swift

```diff
@@ -250,6 +260,68 @@ public final class BuildOperation: PackageStructureDelegate, SPMBuildCore.BuildS
         }
     }
 
+    private static var didEmitUnexpressedDependencies = false
+
+    private func detectUnexpressedDependencies() {
+        // Ensure we only emit these once, regardless of how many builds are being done.
+        guard !Self.didEmitUnexpressedDependencies else {
+            return
+        }
+        Self.didEmitUnexpressedDependencies = true
+
+        let availableFrameworks = Dictionary<String, PackageIdentity>(uniqueKeysWithValues: AvailableLibraries.compactMap {
+            if let identity = Set($0.identities.map(\.identity)).spm_only {
+                return ("\($0.productName!).framework", identity)
+            } else {
+                return nil
+            }
+        })
+
+        buildDescription.targetDependencyMap?.keys.forEach { targetName in
+            let c99name = targetName.spm_mangledToC99ExtendedIdentifier()
+            // Since we're analysing post-facto, we don't know which parameters are the correct ones.
+            let possibleTempsPaths = [productsBuildParameters, toolsBuildParameters].map {
+                $0.buildPath.appending(component: "\(c99name).build")
+            }
+
+            let usedSDKDependencies: [String] = Set(possibleTempsPaths).flatMap {
+                guard let contents = try? self.fileSystem.readFileContents($0.appending(component: "\(c99name).d")) else {
+                    return [String]()
+                }
+
+                // FIXME: We need a real makefile deps parser here...
+                let deps = contents.description.split(whereSeparator: { $0.isWhitespace })
+                return deps.filter {
+                    $0.contains("Developer/Platforms/MacOSX.platform") // this is our proxy for "library from the SDK"
+                }.compactMap {
+                    try? AbsolutePath(validating: String($0))
+                }.compactMap {
+                    // FIXME: Obviously this is macOS specific
+                    $0.components.first(where: { $0.hasSuffix(".framework") })
```

Comment:
This is probably fine since we initially only anticipate using this on macOS, but maybe refactor into a method of sorts

Reply:
We're utilizing `.framework` in two places, so I decided to leave this as-is for now and dropped a TODO for the overarching problem here.